src/components/LocaleSelector.vue

```diff
@@ -37,12 +40,18 @@ export default {
   },
   data() {
     return {
-      langs: Object.keys(locales),
+      locales,
     };
   },
   methods: {
-    getLanguageName(lang) {
-      return locales[lang];
+    updateRouter({ target: { value: currentLocale } }) {
+      this.$i18n.locale = currentLocale;
```

Comment:
Do we need to worry about the "Component vs. root scope" warning from this page? https://kazupon.github.io/vue-i18n/guide/locale.html

Reply:
That's interesting. I've been trying and both `this.$i18n.locale` and `this.$root.$i18n.locale` work fine... 🤔 Do you see any issue without the $root?

Reply:
🤷‍♂️